ddl/db_partition_test.go

```diff
@@ -3505,6 +3506,255 @@ func TestPartitionErrorCode(t *testing.T) {
 	tk1.MustExec("commit")
 }
 
+func TestCoalescePartition(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	// coalesce partition
+	tk.MustExec(`create schema coalescePart`)
+	tk.MustExec(`use coalescePart`)
+	tk.MustExec(`create table t (
+		id int,
+		fname varchar(30),
+		lname varchar(30),
+		signed date
+	)
+	partition by hash( month(signed) )
+	partitions 12`)
+	for i := 0; i < 200; i++ {
+		tk.MustExec(`insert into t values (?, "Joe", "Doe", from_days(738974 + ?))`, i, i*3)
+	}
+	tk.MustExec("alter table t coalesce partition 4")
+	tk.MustQuery(`show create table t`).Check(testkit.Rows("" +
+		"t CREATE TABLE `t` (\n" +
+		"  `id` int(11) DEFAULT NULL,\n" +
+		"  `fname` varchar(30) DEFAULT NULL,\n" +
+		"  `lname` varchar(30) DEFAULT NULL,\n" +
+		"  `signed` date DEFAULT NULL\n" +
+		") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin\n" +
+		"PARTITION BY HASH (MONTH(`signed`)) PARTITIONS 8"))
+	tk.MustExec(`analyze table t`)
+	tk.MustQuery(`select partition_name, table_rows from information_schema.partitions where table_name = 't' and table_schema = 'coalescePart'`).Sort().Check(testkit.Rows(""+
```

Comment:
The data of `table_rows` in `information_schema.partitions` is read from statistic, and it maybe inaccurate, we should use `select * from t partition (p1);` to check each partitions rows?

Reply:
Should the data not be deterministic and at least aproximate? I can do a `SELECT` as well, but would like to keep the statistics there, to have it covered